x-pack/plugins/infra/public/pages/metrics/index.tsx

```diff
@@ -89,7 +89,7 @@ export const MetricDetail = withTheme(
         const layouts = layoutCreator(this.props.theme);
 
         return (
-          <MetricsTimeContainer.Provider>
+          <MetricDetailPageProviders>
```

Comment:
@Kerry350 I extracted the providers into a separate component, because the nesting became a bit excessive after merging in the current `master`. What do you think?

Reply:
Looks good 👍 I'm liking the `page_providers` pattern in general (from here and in the original changes).